### PR TITLE
feat(DataSpreadsheet): add keyboard support to cell selections

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #1717 

This PR allows for cell selections via keyboard navigation. To do this, you hold the shift key, together with an arrow key and a cell selection area will be created.

https://user-images.githubusercontent.com/10215203/156415329-2e13db83-0d1d-498e-90e3-65b74b54c0a1.mov

#### What did you change?
`DataSpreadsheet.js`
#### How did you test and verify your work?
Storybook

### CODE REVIEW

File: packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js [hunk 1]
```diff
@@ -245,119 +327,141 @@ export let DataSpreadsheet = React.forwardRef(
...
+          handleMultipleKeys();
+        }
+        // Allow arrow key navigation if there are less than two activeKeys OR
+        // if one of the activeCellCoordinates is in a header position
+        if (
+          !activeKeys.current.includes(16) ||
+          activeCellCoordinates.row === 'header' ||
+          activeCellCoordinates.column === 'header'
+        ) {
+          switch (keyCode) {
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js:356]: No changes were made to this switch statement, I just wrapped it in the if statement above which made github highlight it as a bunch of changes

Comment (elycheea) [reply packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js:356]: Reviewer can set hide whitespace changes in settings in that case. 😄

Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js:356]: Cool! I did not know that

**Review by lee-chase — APPROVED** [review]
I'll approve Matt but a follow up PR removing the use of keyCode is needed.

File: packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js [hunk 2]
```diff
@@ -429,6 +533,18 @@ export let DataSpreadsheet = React.forwardRef(
       containerHasFocus,
     ]);
 
+    const handleKeyUp = (event) => {
+      const { keyCode } = event;
```
[the remark this answers is not in the document]
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js:537]: Updated here https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/1727

**Pull request merged**